### PR TITLE
feat: fiat -> userCurrency vernacular in swapper domain

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

See https://github.com/shapeshift/web/pull/4882#discussion_r1252382804

This PR:

1. Makes it obvious when we're selecting the "fiat" rate/price where fiat is actually programmatic (user-selected currency)
2. By extension, now makes it obvious when we're selecting USD explicitly, since we will have to use the USD vernacular in contrast with "userCurrency"

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Relates to https://github.com/shapeshift/web/issues/4866, vernacular groundwork which will make it easier to fix where we've goof'd up

## Risk

Effectively none, this is only changing variable names

## Testing

N/A

### Engineering

- Vernacular makes sense

### Operations

Nothing to see here

### CODE REVIEW

**Comment by gomesalexandre** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #4892** <a href="https://app.graphite.dev/github/pr/shapeshift/web/4892" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4892?utm_source=stack-comment).

File: src/components/MultiHopTrade/helpers.ts [hunk 1]
```diff
@@ -36,7 +36,7 @@ export const getMixpanelEventData = () => {
   return {
     buyAsset: compositeBuyAsset,
     sellAsset: compositeSellAsset,
-    fiatAmount: sellAmountBeforeFeesFiat,
+    amountUSD: sellAmountBeforeFeesUserCurrency,
```
Comment (gomesalexandre) [inline src/components/MultiHopTrade/helpers.ts:39]: revert, this is "user" facing and should be in a separate PR for granularity

Comment (gomesalexandre) [reply src/components/MultiHopTrade/helpers.ts]: https://github.com/shapeshift/web/pull/4892/commits/95661a9fe9972bd5668a21250647afc37d1efb73

File: src/components/Approval/Approval.tsx [hunk 2]
```diff
@@ -62,7 +62,7 @@ export const Approval = () => {
   } = useFormContext()
 
   const activeQuote = useSwapperStore(selectQuote)
-  const feeAssetFiatRate = useSwapperStore(selectFeeAssetFiatRate)
+  const feeAssetFiatRate = useSwapperStore(selectFeeAssetUserCurrencyRate)
```
Comment (kaladinlight) [inline src/components/Approval/Approval.tsx:65]: seeing quite a few variable names that weren't updated with the same vernacular if we want to keep that in alignment

Comment (gomesalexandre) [reply src/components/Approval/Approval.tsx]: Think I got most in the commits following this comment, hard to cover it all but at least the main ones should be tackled here - we can probably tackle the rest as we see them

**Review by kaladinlight — APPROVED** [review]

**Comment by gomesalexandre** [comment]
@gomesalexandre started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4892).

**Comment by gomesalexandre** [comment]
@gomesalexandre merged this pull request with [Graphite](https://app.graphite.dev/github/pr/shapeshift/web/4892).
